rtwo/drivers/openstack_user.py

```diff
@@ -37,25 +40,34 @@ def lc_driver_init(self, lc_driver, *args, **kwargs):
         return manager
 
     def __init__(self, *args, **kwargs):
+        self.built_using_token=False
         self.keystone, self.nova, self.swift = self.new_connection(*args, **kwargs)
-        auth_version = kwargs.get('version','v2.0')
-        if '2.0' in auth_version:
-            self.version = 2
-        elif '3' in auth_version:
-            self.version = 3
-        else:
-            raise Exception("Could not determine a version of openstack_user based on %s" % auth_version)
-
+        if not kwargs.has_key('ex_force_auth_token'): 
+            auth_version = kwargs.get('version','v2.0')
+            if '2.0' in auth_version:
+                self.version = 2
+            elif '3' in auth_version:
+                self.version = 3
+            else:
+                raise Exception("Could not determine a version of openstack_user based on %s" % auth_version)
 
     def new_connection(self, *args, **kwargs):
-        keystone = _connect_to_keystone(*args, **kwargs)
-        nova_args = kwargs.copy()
-        #HACK - Nova is certified-broken-on-v3. 
-        nova_args['version'] = 'v2.0'
-        nova_args['auth_url'] = nova_args['auth_url'].replace('v3','v2.0')
-        nova = _connect_to_nova(*args, **nova_args)
-        swift_args = self._get_swift_args(*args, **kwargs)
-        swift = _connect_to_swift(*args, **swift_args)
+        if kwargs.has_key('ex_force_auth_token'):
+            keystone = _connect_to_keystone(*args, **kwargs)
+            nova_args = kwargs.copy()
+            #HACK - Nova is certified-broken-on-v3. 
+            nova_args['version'] = 'v2.0'
+            nova_args['auth_url'] = nova_args['auth_url'].replace('v3','v2.0')
+            nova = _connect_to_nova(*args, **nova_args)
+            swift_args = self._get_swift_args(*args, **kwargs)
+            swift = _connect_to_swift(*args, **swift_args)
+        else:
```

Comment:
Refactor this -- get rid of if/else logic and let `_connect_to_*` handle the kwargs independently.